pkg/landscaper/installations/helper.go

```diff
@@ -449,3 +450,70 @@ func ResolveConfigMapReference(ctx context.Context, kubeClient client.Client, co
 
 	return completeData, data, cm.Generation, nil
 }
+
+// HandleSubComponentPhaseChanges updates the phase of the given installation, if its phase doesn't match the combined phase of its subinstallations/executions anymore
+func HandleSubComponentPhaseChanges(
+	ctx context.Context,
+	kubeClient client.Client,
+	inst *lsv1alpha1.Installation,
+	getOperation func(context.Context, *lsv1alpha1.Installation) (*Operation, error),
+	getExports func(context.Context, *Operation) ([]*dataobjects.DataObject, []*dataobjects.Target, error),
+	logger logr.Logger) error {
```

Comment:
nit: in other functions we pass the logger as second argument after the context.
```suggestion
	ctx context.Context,
	log logr.Logger,
	kubeClient client.Client,
	inst *lsv1alpha1.Installation,
	getOperation func(context.Context, *lsv1alpha1.Installation) (*Operation, error),
	getExports func(context.Context, *Operation) ([]*dataobjects.DataObject, []*dataobjects.Target, error)) error {
```

Reply:
Ah, I wasn't sure about that. I changed it. Will squash the commits during merge.